STEP/CodeGeneration/CodeGenerationVisitor.cs

```diff
@@ -29,7 +34,19 @@ private void EmitAppend(string line)
     private void EmitAppend(TypeVal typeVal)
     {
         // TODO: translate our types to Arduino types here
-        _stringBuilder.Append(typeVal.ToString()+" ");
+        switch(typeVal)
+        {
+            case TypeVal.String:
+                _stringBuilder.Append("String ");
```

Comment:
string med stort men ikke bool og double, er det bare sådan arduino er ?

Reply:
Yes - der er en `string` datatype i Arduino/C++, men den er meget basic (stort set det samme som et char array som i gode gamle C). 
`String` (med stort S) er en klasse de selv har lavet til Arduino, hvor de har lavet operator overloading for string concatenation, equality, etc.
Der står lidt om det her: https://www.arduino.cc/reference/en/language/variables/data-types/string/

Reply:
oright, det sprang bare i øjenene, men epic nok, måske en comment jeg ville aldrig have gættet det :P